sql/moz-fx-data-shared-prod/telemetry_derived/newtab_conditional_daily_aggregates_v1/query.sql

```diff
@@ -0,0 +1,171 @@
+  ) AS sponsored_topsite_impressed_client_count,
+  COUNT(
+    DISTINCT IF(channel = 'release' AND sponsored_topsite_impressions > 0, newtab_visit_id, NULL)
+  ) AS sponsored_topsite_impressed_visit_count,
+  COUNT(
+    DISTINCT IF(channel = 'release' AND sponsored_topsite_clicks > 0, client_id, NULL)
+  ) AS sponsored_topsite_clicked_client_count,
+  COUNT(
+    DISTINCT IF(channel = 'release' AND sponsored_topsite_clicks > 0, newtab_visit_id, NULL)
+  ) AS sponsored_topsite_clicked_visit_count,
+  COUNT(
+    DISTINCT IF(channel = 'release' AND topsite_impressions > 0, client_id, NULL)
+  ) AS topsite_impressed_client_count,
+  COUNT(
+    DISTINCT IF(channel = 'release' AND topsite_impressions > 0, newtab_visit_id, NULL)
+  ) AS topsite_impressed_visit_count,
+  COUNT(
+    DISTINCT IF(channel = 'release' AND topsite_clicks > 0, client_id, NULL)
+  ) AS topsite_clicked_client_count,
+  COUNT(
+    DISTINCT IF(channel = 'release' AND topsite_clicks > 0, newtab_visit_id, NULL)
+  ) AS topsite_clicked_visit_count,
+  COUNT(
+    DISTINCT IF(channel = 'release' AND sponsored_pocket_impressions > 0, client_id, NULL)
+  ) AS sponsored_pocket_impressed_client_count,
+  COUNT(
+    DISTINCT IF(channel = 'release' AND sponsored_pocket_impressions > 0, newtab_visit_id, NULL)
+  ) AS sponsored_pocket_impressed_visit_count,
+  COUNT(
+    DISTINCT IF(channel = 'release' AND sponsored_pocket_clicks > 0, client_id, NULL)
+  ) AS sponsored_pocket_clicked_client_count,
+  COUNT(
+    DISTINCT IF(channel = 'release' AND sponsored_pocket_clicks > 0, newtab_visit_id, NULL)
+  ) AS sponsored_pocket_clicked_visit_count,
+  COUNT(
+    DISTINCT IF(channel = 'release' AND organic_pocket_impressions > 0, client_id, NULL)
+  ) AS organic_pocket_impressed_client_count,
+  COUNT(
+    DISTINCT IF(channel = 'release' AND organic_pocket_impressions > 0, newtab_visit_id, NULL)
+  ) AS organic_pocket_impressed_visit_count,
+  COUNT(
+    DISTINCT IF(channel = 'release' AND organic_pocket_clicks > 0, client_id, NULL)
+  ) AS organic_pocket_clicked_client_count,
+  COUNT(
+    DISTINCT IF(channel = 'release' AND organic_pocket_clicks > 0, newtab_visit_id, NULL)
+  ) AS organic_pocket_clicked_visit_count,
+  COUNT(
+    DISTINCT IF(channel = 'release' AND pocket_impressions > 0, client_id, NULL)
+  ) AS pocket_impressed_client_count,
+  COUNT(
+    DISTINCT IF(channel = 'release' AND pocket_impressions > 0, newtab_visit_id, NULL)
+  ) AS pocket_impressed_visit_count,
+  COUNT(
+    DISTINCT IF(channel = 'release' AND pocket_clicks > 0, client_id, NULL)
+  ) AS pocket_clicked_client_count,
+  COUNT(
+    DISTINCT IF(channel = 'release' AND pocket_clicks > 0, newtab_visit_id, NULL)
+  ) AS pocket_clicked_visit_count,
+  COUNT(
+    DISTINCT IF(channel = 'release' AND pocket_is_signed_in, client_id, NULL)
+  ) AS pocket_signed_in_client_count,
+  COUNT(
+    DISTINCT IF(channel = 'release' AND pocket_is_signed_in, newtab_visit_id, NULL)
+  ) AS pocket_signed_in_visit_count,
```

Comment:
Do we need one for counting just homepage category enabled/disabled independent of source?
Maybe I missed it but I couldn't find.

For this widget: https://mozilla.cloud.looker.com/explore/firefox_desktop/newtab_interactions?qid=qOIDEWK6Oo10MxPDZcg7rn&origin_space=464&toggle=vis

Reply:
Also for this widget with newtab_category: https://mozilla.cloud.looker.com/explore/firefox_desktop/newtab_interactions?qid=L3HcgEzCJWLQBVDaYeVVbg&origin_space=464&toggle=vis

I think due to the count distinct we can't just SUM home vs newtab in the end.

Reply:
I have it here:
- [homepage category enabled](https://github.com/mozilla/bigquery-etl/blob/793aa7b296a535d1a78718a5231fd423e26bd95a/sql/moz-fx-data-shared-prod/telemetry_derived/newtab_conditional_daily_aggregates_v1/query.sql#L12-L29)
-  [homepage category disabled](https://github.com/mozilla/bigquery-etl/blob/793aa7b296a535d1a78718a5231fd423e26bd95a/sql/moz-fx-data-shared-prod/telemetry_derived/newtab_conditional_daily_aggregates_v1/query.sql#L48-L65)

Would this work? Or perhaps I could add `newtab_homepage_category` and/or `newtab_newtab_category` to the granularity (GROUP BY) if you feel its ok (that is: we will not end up double counting clients and visits

Reply:
My point was that all aggregations of that field have either `newtab_open_source = 'about:home'` or `newtab_open_source = 'about:newtab'` but the query on the explore doesn't include anything about newtab_open_source and I'm not sure how that affects the numbers.

Reply:
Ok. Just to be safe, instead of adding `newtab_homepage_category` and `newtab_newtab_category` as a dimension in the GROUP BY, I will add a set of 2 distinct count metrics:
- one set for `newtab_homepage_category` enabled
- one set for `newtab_newtab_category` enabled

Reply:
@bani added the [aggregates](https://github.com/mozilla/bigquery-etl/blob/fa5746a5da3f8481cfe83ca399bdabb8736250af/sql/moz-fx-data-shared-prod/telemetry_derived/newtab_conditional_daily_aggregates_v1/query.sql#L12-L43) for newtab_homepage_category enabled and newtab_newtab_category enabled